app/Libraries/Search/BeatmapsetSearchRequestParams.php

```diff
@@ -274,21 +283,23 @@ private function parseQuery(): void
     private function parseSortOrder(?string $value)
     {
         $array = explode('_', $value);
-        $field = static::remapSortField($array[0]);
-        $order = $array[1] ?? null;
+        $this->sortField = $array[0];
+        $this->sortOrder = $array[1] ?? null;
 
-        if (!in_array($order, ['asc', 'desc'], true)) {
-            $order = 'desc';
+        $mappedField = static::remapSortField($this->sortField);
+        if ($mappedField === null) {
+            $this->sortField = $this->getDefaultSortField();
+            $mappedField = static::remapSortField($this->sortField);
         }
 
-        if (empty($field)) {
-            $this->sorts = $this->getDefaultSort($order);
-        } else {
-            $this->sorts = $this->normalizeSort(new Sort($field, $order));
+        if (!in_array($this->sortOrder, ['asc', 'desc'], true)) {
+            $this->sortOrder = 'desc';
         }
 
+        $this->sorts = $this->normalizeSort(new Sort($mappedField, $this->sortOrder));
```

Comment:
maybe make `remapSortField` non-static (and rename?) and assign `$this->sortField` there? The way it bounces between `$mappedField` and `$this->sortField` (and then `sortOrder` in the middle) reads a bit confusing.

Reply:
doing that in the `normalizeSort` (now `setSorts`) seems like a better idea :man_dancing: